src/components/workflows-configuration.js

```diff
@@ -0,0 +1,584 @@
+                    </Grid>
+                </Grid>
+            ))}
+            <Grid container item xs={12} sm={6}>
+                <Button
+                    className={classes.addNewArea}
+                    variant="outlined"
+                    onClick={() => handleAddArea()}
+                >
+                    <AddCircleIcon
+                        fontSize="default"
+                        style={{ marginRight: '10px' }}
+                    />
+                    <FormattedMessage id="addNewProcess" />
+                </Button>
+            </Grid>
+        </Grid>
+    );
+};
+
+const WorkflowsConfiguration = ({ open, onClose }) => {
+    const [areasWorkFlows, setAreasWorkFlows] = useState([]);
+    const configs = useSelector((state) => state.configs);
+    const [confirmSave, setConfirmSave] = useState(false);
+    const dispatch = useDispatch();
+
+    useEffect(() => {
+        setConfirmSave(false);
+    }, [open]);
+
+    const handleClose = () => {
+        onClose();
+    };
+
+    const handleClosePopup = () => {
+        setConfirmSave(false);
+    };
+
+    const saveButtonClicked = () => {
+        if (processesToBeDeleted().length === 0) {
+            handleSave();
+        } else {
+            setConfirmSave(true);
+        }
+    };
+
+    const processesToBeDeleted = useCallback(() => {
+        const initialProcesses = configs.map((e) => e.process);
+        const currentProcesses = areasWorkFlows
+            .filter((e) => e.process !== '')
+            .map((e) => e.process);
+
+        let toBeDeleted = [];
+        // PROCESSES
+        for (let i = 0; i < initialProcesses.length; i++) {
+            if (!currentProcesses.includes(initialProcesses[i])) {
+                toBeDeleted.push(initialProcesses[i]);
+            }
+        }
+        return toBeDeleted;
+    }, [configs, areasWorkFlows]);
+
+    const handleSave = () => {
+        const initialProcesses = configs.map((e) => e.process);
+        const currentProcesses = areasWorkFlows
+            .filter((e) => e.process !== '')
+            .map((e) => e.process);
+
+        let promises = [];
+        let i = 0;
+
+        // DELETE PROCESSES
+        for (i; i < initialProcesses.length; i++) {
+            if (!currentProcesses.includes(initialProcesses[i])) {
+                promises.push(deleteProcess(initialProcesses[i]));
+            }
+        }
+
+        const WorkFlowWithoutId = (workflow) => {
+            return {
+                process: workflow.process,
+                runBalancesAdjustment: workflow.runBalancesAdjustment,
+                tsos: workflow.tsos.map((tso) => {
+                    return {
+                        sourcingActor: tso.sourcingActor,
+                        alternativeSourcingActor: tso.alternativeSourcingActor,
+                    };
+                }),
+            };
+        };
+
+        const areDifferent = (initialWorkflow, areaWorkFlow) => {
+            let isDifferent = false;
```

Comment:
seems complicated, can you try to simplify ? For example, when there are no duplicates, you can just check that the lengths are equal after the first forEach. I think there are other ways to simplify this